Error compiling with gcc 12.1.0 in linux

compiling with gcc 12.1.0 in arch linux I got this error in b507ba6f2f719c082b79bdfd2dc7c4b3338980a7
---

[ 73%] Built target OgreTerrain
[ 73%] Building CXX object PlugIns/DotScene/CMakeFiles/Plugin_DotScene.dir/src/DotSceneLoader.cpp.o
[ 73%] Linking CXX shared library ../../lib/Plugin_DotScene.so
[ 73%] Built target Plugin_DotScene
[ 73%] Building CXX object PlugIns/Assimp/CMakeFiles/Codec_Assimp.dir/src/AssimpLoader.cpp.o
/home/joilnen/ogre3d_devel/ogre3d-13.4.x/PlugIns/Assimp/src/AssimpLoader.cpp: In member function ‘void Ogre::AssimpLoader::grabBoneNamesFromNode(const aiScene*, const aiNode*)’:
/home/joilnen/ogre3d_devel/ogre3d-13.4.x/PlugIns/Assimp/src/AssimpLoader.cpp:818:50: error: comparison between two arrays [-Werror=array-compare]
  818 |                             if (node->mName.data == pNode->mName.data)
      |                                 ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
/home/joilnen/ogre3d_devel/ogre3d-13.4.x/PlugIns/Assimp/src/AssimpLoader.cpp:818:50: note: use unary ‘+’ which decays operands to pointers or ‘&‘component_ref’ not supported by dump_decl<declaration error>[0] == &‘component_ref’ not supported by dump_decl<declaration error>[0]’ to compare the addresses
/home/joilnen/ogre3d_devel/ogre3d-13.4.x/PlugIns/Assimp/src/AssimpLoader.cpp:823:50: error: comparison between two arrays [-Werror=array-compare]
  823 |                             if (node->mName.data == pNode->mParent->mName.data)
      |                                 ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joilnen/ogre3d_devel/ogre3d-13.4.x/PlugIns/Assimp/src/AssimpLoader.cpp:823:50: note: use unary ‘+’ which decays operands to pointers or ‘&‘component_ref’ not supported by dump_decl<declaration error>[0] == &‘component_ref’ not supported by dump_decl<declaration error>[0]’ to compare the addresses
cc1plus: all warnings being treated as errors
make[2]: *** [PlugIns/Assimp/CMakeFiles/Codec_Assimp.dir/build.make:76: PlugIns/Assimp/CMakeFiles/Codec_Assimp.dir/src/AssimpLoader.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1967: PlugIns/Assimp/CMakeFiles/Codec_Assimp.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

--- a/PlugIns/Assimp/src/AssimpLoader.cpp
+++ b/PlugIns/Assimp/src/AssimpLoader.cpp
@@ -367,6 +367,10 @@ bool AssimpLoader::_load(const char* name, Assimp::Importer& importer, Mesh* mes
         flags |= aiProcess_JoinIdenticalVertices;
 
     flags |= options.postProcessSteps;
+
+    if((flags & (aiProcess_GenSmoothNormals | aiProcess_GenNormals)) != aiProcess_GenNormals)
+        flags &= ~aiProcess_GenNormals; // prefer smooth normals
+
     importer.SetPropertyFloat("PP_GSN_MAX_SMOOTHING_ANGLE", options.maxEdgeAngle);
     const aiScene* scene = importer.ReadFile(name, flags);
 

--- a/PlugIns/Assimp/src/AssimpLoader.cpp
+++ b/PlugIns/Assimp/src/AssimpLoader.cpp
@@ -819,12 +819,12 @@ void AssimpLoader::grabBoneNamesFromNode(const aiScene* mScene, const aiNode* pN
                         aiNode* node = mScene->mRootNode->FindNode(pAIBone->mName.data);
                         while (node)
                         {
-                            if (node->mName.data == pNode->mName.data)
+                            if (node->mName == pNode->mName)
                             {
                                 flagNodeAsNeeded(node->mName.data);
                                 break;
                             }
-                            if (node->mName.data == pNode->mParent->mName.data)
+                            if (node->mName == pNode->mParent->mName)
                             {
                                 flagNodeAsNeeded(node->mName.data);
                                 break;

